chainerx_cc/chainerx/array_body.cc

```diff
@@ -141,7 +139,10 @@ ReturnType ArrayBody::GetGradImpl(ThisPtr this_ptr, const BackpropId& backprop_i
         return nullptr;
     }
     CHAINERX_ASSERT(*i < this_ptr->grads_.size());
-    return this_ptr->grads_[*i].get();
+    ReturnType grad = this_ptr->grads_[*i].get();
+    CHAINERX_ASSERT(grad != nullptr);
+    CHAINERX_ASSERT(GetKind(this_ptr->dtype_) == DtypeKind::kFloat || !grad->has_value());  // Integral arrays cannot have gradients.
```

Comment:
This is be violated with the following code since it's possible to assign gradients to integral arrays. Should it?
```python
a = chx.arange(3).require_grad()
a.grad = chx.arange(3)  
a.grad  # Assertion failure
```

Reply:
Fixed and updated a test to check integer arrays.